Spiffe provider

gtcooke94 opened the pull request:
This PR provides a diffbase for SPIFFE bundles to be added to the root provider key material

easwars committed:
cds: stop child policies on resource-not-found errors (#8122)

easwars committed:
xds: simplify code handling certain error conditions in the resolver (#8123)

arjan-bal committed:
xds, pickfirst: Enable additional addresses in xDS, set new pick_first as default (#8126)

arjan-bal committed:
github: change test action to cover the legacy pickfirst balancer (#8129)

gtcooke94 committed:
more watcher tests

gtcooke94 committed:
add another test file

gtcooke94 committed:
rebase conflict

janardhanvissa committed:
cleanup: replace dial with newclient (#7967)

janardhanvissa committed:
cleanup: replace dial with newclient (#7970)

purnesh42H committed:
stats/openetelemetry: refactor and make e2e test stats verification deterministic (#8077)

purnesh42H committed:
xds: introduce simple grpc transport for generic xds clients (#8066)

arjan-bal committed:
transport: Send RST stream from the server when deadline expires (#8071)

arjan-bal committed:
clusterresolver: Lower log level when ExitIdle is called with no child (#8133)

easwars committed:
xds: ensure xDS node ID is populated in errors from xds resolver and cds lb policy (#8131)

arjan-bal committed:
interop: Wait for server to become ready in alts interop tests (#8141)

arjan-bal committed:
*: Regenerate protos (#8142)

arjan-bal committed:
priority: Send and validate connection error in test (#8143)

purnesh42H committed:
xds: introduce generic xds clients xDS and LRS Client API signatures (#8042)

chressie committed:
balancer/pickfirst/pickfirstleaf: fix race condition in tests (#8148)

arjan-bal committed:
grpc: Add endpoints in resolverWrapper.NewAddresses (#8149)

gtcooke94 committed:
merge

gtcooke94 committed:
modify to new APIs

gtcooke94 committed:
mod tidy

gtcooke94 committed:
fix test expectation

dfawley committed:
resolver/manual: allow calling UpdateState with an un-Built resolver (#8150)

gtcooke94 committed:
slight cleanup

gtcooke94 committed:
merge spiffe

gtcooke94 committed:
mod tidy

gtcooke94 committed:
Merge branch 'spiffe' into spiffe_provider

gtcooke94 committed:
merge

gtcooke94 committed:
mis-merged

gtcooke94 committed:
fix merge issues

matthewstevenson88 commented on credentials/tls/certprovider/pemfile/watcher.go:
```diff
@@ -191,23 +198,57 @@ func (w *watcher) updateRootDistributor() {
 		return
 	}
 
+	// Prefer SPIFFEBundleMap, if it works don't use the RootFile
```
micro-nit: Missing period at the end of the sentence.

matthewstevenson88 commented on credentials/tls/certprovider/pemfile/watcher.go:
```diff
@@ -191,23 +198,57 @@ func (w *watcher) updateRootDistributor() {
+	}
+	spiffeBundleMapContents, err := os.ReadFile(w.opts.SPIFFEBundleMapFile)
+	if err != nil {
+		logger.Warningf("spiffeBundleMapFile (%s) read failed: %v", w.opts.SPIFFEBundleMapFile, err)
```
Do we really want to log an error here and below? Should whoever calls this function be logging instead?

gtcooke94 replied:
I believe so (and this structure is copied from the Root Cert reloading). No one is intentionally calling this function, rather it runs periodically in the background to reload the data from a file, so the error wouldn't get passed back out to the main thread.

matthewstevenson88 replied:
SG, as long as we are not double logging, I'm fine. :)

LGTM.

matthewstevenson88 reviewed (approved):
Two small comments, but no concerns with this PR.

easwars committed:
xds: ensure node ID is populated in errors from the server (#8140)

* xds: ensure node ID is populated in errors from the server

* xds: cleanup server e2e tests

* don't wrap status errors

gtcooke94 committed:
add period in comment

gtcooke94 committed:
change behavior to not fallback to root file

easwars committed:
xds: emit resource-not-found logs at Warning level (#8158)

shivaspeaks committed:
balancer/rls: allow maxAge to exceed 5m if staleAge is set (#8137)

gtcooke94 committed:
Merge branch 'spiffe' into spiffe_provider

gtcooke94 committed:
[Security] Add support for parsing SPIFFE Bundle Maps (#8124)

This adds a dependency on go-spiffe in order to parse SPIFFE bundles. More specifically, that library does not yet support SPIFFE bundle maps, but it does support SPIFFE bundles. This adds parsing of these maps to grpc-go

gtcooke94 committed:
Merge remote-tracking branch 'upstream/master' into spiffe_provider

gtcooke94 committed:
merge things

gtcooke94 committed:
newline

gtcooke94 committed:
remove stutter

gtcooke94 committed:
cleanup
